conventions/documentation/actionmenus.md

```diff
@@ -4,24 +4,75 @@ parent: Conventions
-- Edit [Resource Name]
-- Delete [Resource Name]
+**Topology side panel**
+![action menus topology panel](../images/action-menus-topology-panel.png)
+
+**Topology right click action on a resource**
+![action menus topology](../images/action-menus-topology.png)
+
+- Actions dropdowns should be styled as primary blue dropdowns.
+- Exceptions are acceptable in cases where a different action(s) has been identified as the view's primary action. In these cases, the primary action(s) is usually pulled out of the dropdown and is styled as a primary button or dropdown to the left of the Actions dropdown, which will thus have secondary styling (e.g., Secrets details pages).
+
+## Organization of Actions menu items
+
+**Basic action menus**
+
+<img src="../images/action-menus-1.png" alt="Action menus order" width="294"/>
+
+The majority of resource action menus will be a basic action menu. These menus have "universal" actions and the `Delete {resource_type}` action separated by a horizontal divider.
+
+**Complex action menus**
+
+Some resource action menus will have more actions than those included in the basic action menu. The following documentation uses a Deployment as an example, but the logic to each section should be applied to any other complex action menu.
+
+<img src="../images/action-menus-2.png" alt="Action menus order" width="1239"/>
+
+Action menus should be separated into multiple sections. The ordering of sections and actions in the menu should adhere to the following pattern:
+1. **Resource specific actions**
+  * These should be ordered with standalone actions listed first followed by flyouts, with each separate group ordered alphabetically by default.
+  * Exceptions are acceptable in the cases where an action may be dynamic (e.g., `Pause rollouts` becomes `Resume rollouts`, but the action should not move from its original position).
+2. **"Universal" actions**
+  * These actions are available for every resource in the console. These should be listed in the following order:
+    1. `Edit labels`
+    2. `Edit annotations`
+    3. `Edit {resource_type}`
+3. **Delete resource action**
+  * The last section of each Actions menu should have the `Delete {resource_type}` action.
+4. **Submenus**
+  * In this example, the resource has two or more actions that are Edit actions, and are thus put into an Edit submenu.
+  * Resources may have more than one submenu.
+  * The default logic for ordering these actions is alphabetical.
+  * A special note on the Autoscale submenu: Resources that have autoscaling actions will have a separate submenu for Autoscale. If the resource doesn't yet have a HorizontalPodAutoscaler, the only action in the submenu will be `Add HorizontalPodAutoscaler`. If one exists already, the Add action will be replaced by `Edit HorizontalPodAutoscaler` and `Delete HorizontalPodAutoscaler`. (In the future, VerticalPodAutoscalers may be added to this submenu, with the same dynamic actions used for HPAs – Add, Edit, and Delete.)
```

Comment:
I would love to generalize this beyond HPAs, but maybe that's not our intent with this convention. Something like 'If multiple (dynamically present?) actions relate directly to one topic, they can go in their own submenu. For example: HPAs.'